using-datapower-in-kubernetes/getting-started-datapower-in-gke.md

```diff
@@ -0,0 +1,271 @@
+You can specify deployments, pods, services, replica sets, etc. declaratively in Kubernetes by writing a
+corresponding configuration file.
+
+In a separate Google Cloud Shell session, we can run `$ kubectl describe deployments` to show the details
+ of the *datapower* deployment we just created.
+ 
+ ![kubectl describe deployments](media/kubectl_describe_deployments_datapower.png)
+
+Don't worry if not everything from the output makes sense at this point, the key takeaway here is that we can use the `describe` command to 
+show the details of various Kubernetes objects as we will soon see. Additionally, note the section titled "NewReplicaSet" which lists the replica set used by our deployment.
+We can see the details of our replica sets by using the `$ kubectl describe replicasets`.
+
+![kubectl describe replicasets](media/kubectl_describe_replicasets.png)
+
+As we can see in the section "Replicas", this replica set describes our current state of "1 current" running pod
+and our desired state of "1 desired" running pod.
+
+Similarly, we can run the `$ kubectl get pods` command to view the running datapower pod. 
+
+![kubectl get pods](media/kubectl_get_pods.png)
+
+To demonstrate the role of replica sets, lets manually delete the datapower pod.
+To do this, we run: 
+
+```bash
+$ kubectl delete pods <pod_name>
+```
+
+where <pod_name> is the name of the pod as listed in `$kubectl get pods`.
+
+We expect the replica set to note that change of state, our datapower pod being deleted, and promptly start a new pod for us running our datapower image.
+To ensure this, we check `$ kubectl get pods` again which shows a new pod (note the new name) running our datapower image.
+We can can also see this change in the event output of our replica set.
+
+```bash
+$ kubectl describe replicasets
+```
+
+Lastly, our first Google Cloud Session will also show our previous interactive session with the container has now
+been terminated since we issued a deletion of the pod. To reattach to the new container created by the replica set,
+we can issue:
+
+```bash
+$ kubectl attach <pod_name> --container="datapower" --stdin --tty
+```
+
+And log in to datapower using the default user "admin" and "admin" password.
+
+Once you are done you can delete the deployment as we will not make further use of it in this tutorial.
+
+# Configuration
+
+One best practice is to separate your application code from your configuration. To this end,
+Kubernetes provides a `ConfigMap` resource as of version 1.2.
+A ConfigMap is a flexible object for providing your application with non-secret configuration data. The data
+is presented as key-value pairs and can represent cli arguments, environment variables, and files in a mounted volume.
+
+Let's create a ConfigMap from our datapower/auto-startup.cfg file which looks like this:
+
+```
+top; co
+
+web-mgmt
+    admin enabled
+    port 9090
+exit
+```
```

Comment:
This is encouraging bad behavior and demonstrating an antipattern. I think that the demo should be of some kind of actual application, not just bringing up `web-mgmt`. Remember, we don't think people should be using web-mgmt in prod anyway, and we're not showing them how to do dev, and you've already gone past the point of showing how to run a single container.

I think you should just reuse the hello-world application that we've used elsewhere. That will give you 3 different containers and let them be orchestrated together.

I'll stop the review here until we've made those changes.